File: ios/Sources/ForgeIOSRuntime/NativeDraftState.swift
import Foundation

public struct NativeDraftState: Sendable, Equatable {
    public var baseline: [String: JSONValue]?
    public init(baseline: [String: JSONValue]? = nil) { self.baseline = baseline }
    public func dirty(_ form: [String: JSONValue]) -> Bool { form != (baseline ?? form) }
    public func resetValue(_ form: [String: JSONValue]) -> [String: JSONValue] { baseline ?? form }
    public static func submitExtras(_ form: [String: JSONValue]) -> [String: JSONValue] { ["data": .object(form)] }
    public func resetExtras(_ form: [String: JSONValue]) -> [String: JSONValue] { ["values": .object(resetValue(form))] }
}

File: android/sdk/src/main/java/com/viant/forgeandroid/runtime/NativeDraftState.kt
package com.viant.forgeandroid.runtime

data class NativeDraftState(val baseline: Map<String, Any?>) {
    fun dirty(form: Map<String, Any?>) = form != baseline
    fun accepted(form: Map<String, Any?>) = copy(baseline = form.toMap())
    fun submitExtras(form: Map<String, Any?>) = mapOf("data" to form)
    fun resetExtras() = mapOf("values" to baseline)
}
